source/adapters/opencl/device.cpp

```diff
@@ -938,17 +938,30 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
     return UR_RESULT_SUCCESS;
   }
   case UR_DEVICE_INFO_SUB_GROUP_SIZES_INTEL: {
-    // Have to convert size_t to uint32_t
-    size_t SubGroupSizesSize = 0;
-    CL_RETURN_ON_FAILURE(
-        clGetDeviceInfo(cl_adapter::cast<cl_device_id>(hDevice), CLPropName, 0,
-                        nullptr, &SubGroupSizesSize));
-    std::vector<size_t> SubGroupSizes(SubGroupSizesSize / sizeof(size_t));
-    CL_RETURN_ON_FAILURE(
-        clGetDeviceInfo(cl_adapter::cast<cl_device_id>(hDevice), CLPropName,
-                        SubGroupSizesSize, SubGroupSizes.data(), nullptr));
-    return ReturnValue.template operator()<uint32_t>(SubGroupSizes.data(),
-                                                     SubGroupSizes.size());
+    /* CL_DEVICE_SUB_GROUP_SIZES_INTEL is only supported if the device has the
+     * cl_intel_required_subgroup_size extension, if it does not have it we
+     * return a default subgroups sizes list of {1}
+     * */
+    size_t ExtSize = 0;
+    urDeviceGetInfo(hDevice, UR_DEVICE_INFO_EXTENSIONS, 0, nullptr, &ExtSize);
+    std::string ExtStr(ExtSize, 0);
+    urDeviceGetInfo(hDevice, UR_DEVICE_INFO_EXTENSIONS, ExtSize, ExtStr.data(),
+                    nullptr);
```

Comment:
Using `urDeviceGetInfo` here results in unnecessary indirection. We have the `cl_adapter::checkDeviceExtensions()` helper function which calls OpenCL directly which should be preferred.